hail/src/main/scala/is/hail/expr/ir/LowerMatrixIR.scala

```diff
@@ -369,11 +369,16 @@ object LowerMatrixIR {
       case MatrixDistinctByRow(child) => TableDistinct(lower(child, ab))
 
       case MatrixRowsHead(child, n) => TableHead(lower(child, ab), n)
+      case MatrixRowsTail(child, n) => TableTail(lower(child, ab), n)
 
       case MatrixColsHead(child, n) => lower(child, ab)
         .mapGlobals('global.insertFields(colsField -> 'global (colsField).invoke("[:*]", TArray(child.typ.colType), n)))
         .mapRows('row.insertFields(entriesField -> 'row (entriesField).invoke("[:*]", TArray(child.typ.entryType), n)))
 
+      case MatrixColsTail(child, n) => lower(child, ab)
+        .mapGlobals('global.insertFields(colsField -> 'global (colsField).invoke("[*:]", TArray(child.typ.colType), - n)))
+        .mapRows('row.insertFields(entriesField -> 'row (entriesField).invoke("[*:]", TArray(child.typ.entryType), - n)))
```

Comment:
Can you explain what is going on here to me? Everything looks good to me other than this file, which is probably fine but confuses me because I've never looked at lowering stuff. From pattern matching against head it seems right though. I think if I understood what `invoke("[*:]"` meant that would clear this up enough for me to accept

Reply:
Also have never seen that symbol literal thing before.

Reply:
`invoke(name, ...)` creates an apply IR to call one of the functions in the registry.

`[*:]`, `[:*]`, and `[*:*]` are the functions for slicing. for instance, `[*:*]` gets called when you do an expression like `foo[start:end]` in python to get a subsequence of `foo`. `[:*]` is when you leave out the start index (so it is implicitly `0`), and `[*:]` is when you leave out the end index (so it is implicitly the length of the array). 

if an index is negative, then it means that many steps away from the end of the sequence. for instance `a[-1]` is an idiom in python to get the last element of `a`. so here, i'm doing the slice `[-n:]` to go from `-n` (i.e., `n` steps away from the end) to the end of the array.

```python
>>> "hello"[-2:]
"lo"
```